Assets/Scripts/UI/Visualizers/Patrolling/VisualizationModes/SelectedRobotHighlightingVerticesVisualizationMode.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+// 
+// Contributors: Puvikaran Santhirasegaram
+
+using System.Collections.Generic;
+
+using Accord.Math;
+
+using Maes.Algorithms.Patrolling;
+using Maes.Robot;
+
+using UnityEngine;
+
+namespace Maes.UI.Visualizers.Patrolling.VisualizationModes
+{
+    public class SelectedRobotHighlightingVerticesVisualizationMode : IPatrollingVisualizationMode
+    {
+        public SelectedRobotHighlightingVerticesVisualizationMode(MonaRobot robot)
+        {
+            _robot = robot;
+        }
+
+        private readonly MonaRobot _robot;
+
+        private Dictionary<int, Color32[]> _partitionedVertices = new();
+
+        public void UpdateVisualization(PatrollingVisualizer visualizer, int currentTick)
+        {
+            if (_robot.Algorithm is not IPatrollingAlgorithm alg)
+            {
+                Debug.Log("Selected robot does not have a partitioning algorithm.");
+                return;
+            }
```

Comment:
This check is overkill. You should just cast it directly to IPatrollingAlgorithm.